src/server/ua_server_binary.c

```diff
@@ -519,8 +519,9 @@ processMSG(UA_Server *server, UA_SecureChannel *channel,
         session = &anonymousSession;
     }
 
-    /* Trying to use a non-activated session? */
-    if(sessionRequired && !session->activated) {
+    /* Trying to use a non-activated session?
+     * Do not allow if request is of type CloseSessionRequest */
+    if(sessionRequired && !session->activated && strcmp(requestType->typeName, "CloseSessionRequest")) {
```

Comment:
The strcmp is very indirect logic. Also, there is a build-flag to disable the typenames.
To save ROM on embedded targets. Another solution is
```c
if(sessionRequired && !session->activated && requestType != &UA_TYPES[UA_TYPES_CLOSESESSIONREQUEST])
```
That's also a bit faster.

Reply:
Done!